File: repasoHerenciajava/Profesor.java
import java.util.ArrayList;
public class Profesor extends Personal {
    String gradoAcademic;
    ArrayList<String> materiasDicta;
    public Profesor(String name, String email, String password, String gradoAcademic) {
        super(name, email, password);
        this.gradoAcademic = gradoAcademic;
        this.materiasDicta = new ArrayList<String>();
    }
}

File: repasoHerenciaphp/profesor.php
<?php
require_once("personal.php");
class Profesor extends Personal {
    public $materiasDicta;
    public $nivelAcademico;
    function __construct($name, $email, $password, $materiasDicta){
        parent::__construct($name, $email, $password);
        $this->materiasDicta = $materiasDicta;
    }
}